app/kuma-dp/cmd/run.go

```diff
@@ -137,3 +134,34 @@ func newRunCmd() *cobra.Command {
 	cmd.PersistentFlags().StringVar(&cfg.DataplaneRuntime.TokenPath, "dataplane-token-file", cfg.DataplaneRuntime.TokenPath, "Path to a file with dataplane token (use 'kumactl generate dataplane-token' to get one)")
 	return cmd
 }
+
+// fetchCatalog tries to fetch Kuma CP catalog several times
+// The main reason for introducing retries here is situation when DP is deployed in the same time as CP (ex. Ingress for Remote CP)
+func fetchCatalog(cfg kuma_dp.Config) (*catalog.Catalog, error) {
+	runLog.Info("connecting to the Control Plane API for Bootstrap API location")
+	catalogClient, err := catalogClientFactory(cfg.ControlPlane.ApiServer.URL)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not create catalog client")
+	}
+
+	backoff, err := retry.NewConstant(cfg.ControlPlane.ApiServer.Retry.Backoff)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not create retry backoff")
+	}
+	backoff = retry.WithMaxDuration(cfg.ControlPlane.ApiServer.Retry.MaxDuration, backoff)
+	var c catalog.Catalog
+	err = retry.Do(context.Background(), backoff, func(ctx context.Context) error {
+		c, err = catalogClient.Catalog()
+		if err != nil {
+			runLog.Info("could not connect to the Control Plane API. Retrying.", "backoff", cfg.ControlPlane.ApiServer.Retry.Backoff, "err", err.Error())
```

Comment:
`runLog.Error` will be good.

Reply:
Can you provide a PR against `master`. This is just a cherry-pick for the release of 0.7.2. Thanks!